### PR TITLE
Add Ruby Configuration

### PR DESCRIPTION
**Pull request by rye**
This adds a Ruby configuration for the thing.

### CODE REVIEW

File: .emacs.d/lisp/code/code-hub.el [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 (kotct/hub "code"
            (magit-c
-           indentation
-		   language-hub))
```
Comment (cg505) [inline .emacs.d/lisp/code/code-hub.el:6]: yikes sorry about this

File: .emacs.d/lisp/code/languages/ruby.el [hunk 2]
```diff
@@ -0,0 +1,11 @@
+(setq-default ruby-indent-line global-tab-width)
```
Comment (cg505) [inline .emacs.d/lisp/code/languages/ruby.el:2]: add some comments in this file plz?

**Comment by cg505** [comment]
Looks pretty good except for that one review.

**Review by cg505 — DISMISSED** [review]

File: .emacs.d/lisp/code/languages/ruby.el (hunk 2 above)
Comment (rye) [reply .emacs.d/lisp/code/languages/ruby.el:2]: will do

File: .emacs.d/lisp/code/languages/ruby.el [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+;; Provide advice to Smart Tabs about how to handle Ruby's special
+;; indentation variables.
+(smart-tabs-advice ruby-indent-line ruby-indent-level)
+
+;; Apply Ruby language support to Smart Tabs.
+(smart-tabs-insinuate 'ruby)
+
+;; Use tabs mode in Ruby by default.
+(setf ruby-indent-tabs-mode t)
```
Comment (cg505) [inline .emacs.d/lisp/code/languages/ruby.el:12]: Do we need this in addition to `smart-tabs-insinuate`? We don't in my C config, for instance.

File: .emacs.d/lisp/code/languages/ruby.el (hunk 2 above)
Comment (cg505) [inline .emacs.d/lisp/code/languages/ruby.el]: ty

**Review by cg505 — APPROVED** [review]
ty

File: .emacs.d/lisp/code/languages/ruby.el (hunk 3 above)
Comment (rye) [reply .emacs.d/lisp/code/languages/ruby.el:12]: Yes.  For instances where we have "special" (i.e. arrogant) language modes which disobey `indent-tabs-mode` in favor of their own indent-tabs-mode variables, we need to specify these.

Comment (cg505) [reply .emacs.d/lisp/code/languages/ruby.el:12]: Okay, thank you.

**Comment by cg505** [comment]
Looks good to merge.

**Comment by rye** [comment]
Will merge.

**Pull request merged**